### ISSUE TITLE
[kube-prometheus-stack][prometheus-node-exporter] Node exporter endpoints not working

### ISSUE THREAD

**Issue by zemkogabor** [prometheus-community/helm-charts#1641]
### Describe the bug a clear and concise description of what the bug is.

When I add `endpoints` lines (inside `prometheus-node-exporter` values), I get an error message like this:

```
upgrade.go:430: [debug] warning: Upgrade "monitoring" failed: no Endpoints with the name "monitoring-prometheus-node-exporter" found
Error: UPGRADE FAILED: no Endpoints with the name "monitoring-prometheus-node-exporter" found
helm.go:88: [debug] no Endpoints with the name "monitoring-prometheus-node-exporter" found
```

I tried to delete node exporter service before upgrading, but it didn't help.

### What's your helm version?

version.BuildInfo{Version:"v3.7.2", GitCommit:"663a896f4a815053445eec4153677ddc24a0a361", GitTreeState:"clean", GoVersion:"go1.16.10"}

### What's your kubectl version?

Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.4", GitCommit:"b695d79d4f967c403a96986f1750a35eb75e75f1", GitTreeState:"clean", BuildDate:"2021-11-17T15:48:33Z", GoVersion:"go1.16.10", Compiler:"gc", Platform:"darwin/amd64"} Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.4", GitCommit:"b695d79d4f967c403a96986f1750a35eb75e75f1", GitTreeState:"clean", BuildDate:"2021-11-17T15:42:41Z", GoVersion:"go1.16.10", Compiler:"gc", Platform:"linux/amd64"}

### Which chart?

kube-prometheus-stack

### What's the chart version?

21.0.5

### Enter the changed values of values.yaml?

```
prometheus-node-exporter:
  endpoints:
    - 10.10.21.11
```

### Enter the command that you execute and failing/misfunctioning.

```
helm upgrade monitoring -- install ./chart --namespace monitoring
```

**Comment by dschunack** [comment]
I had a similar issue today after upgrading to helm chart version 25.2.0
Node exporter pod monitor was no longer working and no data available. 
The POD monitor for the exporter in the Sub Chart "prometheus-node-exporter" is false by default.

I added the following lines in my values.yaml file to fix it. Maybe it helps

```yaml
nodeExporter:
  enabled: true
prometheus-node-exporter:
  prometheus:
    monitor:
      enabled: true
```

**Comment by dschunack** [comment]
Nice, it was fixed today with helm chart 26.0.0 👍🏼
